stl/src/vector_algorithms.cpp

```diff
@@ -2930,10 +2930,439 @@ namespace {
+                _mm256_movemask_epi8(_mm256_load_si256(_Table_as_avx + 5)),
+                _mm256_movemask_epi8(_mm256_load_si256(_Table_as_avx + 6)),
+                _mm256_movemask_epi8(_mm256_load_si256(_Table_as_avx + 7)));
+        }
+
+        template <class _Ty>
+        __m256i _Make_bitmap(const _Ty* const _Needle_ptr, const size_t _Needle_length) noexcept {
+            if (_Needle_length <= 20) {
+                return _Make_bitmap_small(_Needle_ptr, _Needle_length);
+            } else {
+                return _Make_bitmap_large(_Needle_ptr, _Needle_length);
+            }
+        }
+    } // namespace __std_find_meow_of_bitmap_details
+#endif // !_M_ARM64EC
+
+    namespace __std_find_meow_of_bitmap {
+#ifndef _M_ARM64EC
+        template <class _Ty>
+        bool _Use_bitmap_avx(const size_t _Count1, const size_t _Count2) noexcept {
+            if constexpr (sizeof(_Ty) == 1) {
+                if (_Count2 <= 16) {
+                    return _Count1 > 1000;
+                } else if (_Count2 <= 48) {
+                    return _Count1 > 80;
+                } else if (_Count2 <= 240) {
+                    return _Count1 > 40;
+                } else if (_Count2 <= 1000) {
+                    return _Count1 > 32;
+                } else {
+                    return _Count1 > 16;
+                }
+            } else if constexpr (sizeof(_Ty) == 2) {
+                if (_Count2 <= 8) {
+                    return _Count1 > 128;
+                } else if (_Count2 <= 48) {
+                    return _Count1 > 32;
+                } else if (_Count2 <= 72) {
+                    return _Count1 > 24;
+                } else if (_Count2 <= 144) {
+                    return _Count1 > 16;
+                } else {
+                    return _Count1 > 8;
+                }
+            } else if constexpr (sizeof(_Ty) == 4) {
+                if (_Count2 <= 8) {
+                    return _Count1 > 64;
+                } else if (_Count2 <= 24) {
+                    return _Count1 > 40;
+                } else if (_Count2 <= 44) {
+                    return _Count1 > 24;
+                } else if (_Count2 <= 112) {
+                    return _Count1 > 16;
+                } else {
+                    return _Count1 > 8;
+                }
+            } else if constexpr (sizeof(_Ty) == 8) {
+                if (_Count2 <= 8) {
+                    return _Count1 > 40;
+                } else if (_Count2 <= 12) {
+                    return _Count1 > 20;
+                } else if (_Count2 <= 48) {
+                    return _Count1 > 16;
+                } else if (_Count2 <= 64) {
+                    return _Count1 > 12;
+                } else if (_Count2 <= 192) {
+                    return _Count1 > 8;
+                } else {
+                    return _Count1 > 4;
+                }
+            } else {
+                static_assert(false, "unexpected size");
+            }
+        }

```

Comment:
`_Use_bitmap_avx` and `_Use_bitmap_scalar` functions are called exactly once each.
I'm not sure if they should be inline or kept separate

Reply:
This is fine.